src/Symfony/Component/Messenger/Transport/Serialization/PhpSerializer.php

```diff
@@ -50,6 +50,14 @@ public function encode(Envelope $envelope): array
         ];
     }
 
+    /**
+     * Entry point to use stricter condition for base64 encoding.
+     */
+    protected function shouldBeEncoded(string $body): bool
```

Comment:
we don't need this extension point: AmazonSqsSerializer can implement the encode method instead, all the parent, and check the result

Reply:
I thought it was better this way but it's changed. I followed your strategy @nicolas-grekas